recipes/boost/all/conanfile.py

```diff
@@ -442,55 +451,18 @@ def validate(self):
+                raise ConanInvalidConfiguration("This compiler is too old to build Boost.nowide.")
 
-        if not all((self.options.without_fiber, self.options.get_safe("without_json", True))):
-            # fiber/json require a c++11-able compiler.
-            if self.settings.compiler.cppstd:
+        if any([not self.options.get_safe(f"without_{library}", True) for library in self._cxx11_boost_libraries]):
+            if self.settings.compiler.get_safe("cppstd"):
                 check_min_cppstd(self, 11)
-            else:
-                version_cxx11_standard = self._min_compiler_version_default_cxx11
-                if version_cxx11_standard:
-                    if Version(self.settings.compiler.version) < version_cxx11_standard:
-                        raise ConanInvalidConfiguration("Boost.{fiber,json} requires a c++11 compiler (please set compiler.cppstd or use a newer compiler)")
-                else:
-                    self.output.warn("I don't know what the default c++ standard of this compiler is. I suppose it supports c++11 by default.\n"
-                                     "This might cause some boost libraries not being built and conan components to fail.")
-
-        if Version(self.version) >= "1.76.0":
-            # Starting from 1.76.0, Boost.Math requires a compiler with c++ standard 11 or higher
-            if not self.options.without_math:
-                if self.settings.compiler.cppstd:
-                    check_min_cppstd(self, 11)
-                else:
-                    min_compiler_version = self._min_compiler_version_default_cxx11
-                    if min_compiler_version is not None:
-                        if Version(self.settings.compiler.version) < min_compiler_version:
-                            raise ConanInvalidConfiguration("Boost.Math requires (boost:)cppstd>=11 (current one is lower)")
-
-        if Version(self.version) >= "1.79.0":
-            # Starting from 1.79.0, Boost.Wave requires a compiler with c++ standard 11 or higher
-            if not self.options.without_wave:
-                if self.settings.compiler.cppstd:
-                    check_min_cppstd(self, 11)
-                else:
-                    min_compiler_version = self._min_compiler_version_default_cxx11
-                    if min_compiler_version is not None:
-                        if Version(self.settings.compiler.version) < min_compiler_version:
-                            raise ConanInvalidConfiguration("Boost.Wave requires (boost:)cppstd>=11 (current one is lower)")
+            version_cxx11_standard = self._min_compiler_version_default_cxx11
+            if version_cxx11_standard and Version(self.settings.compiler.version) < version_cxx11_standard:
+                raise ConanInvalidConfiguration(
+                    f"Boost.{{{','.join(self._cxx11_boost_libraries)}}} requires a c++11 compiler "
+                    "(please set compiler.cppstd or use a newer compiler)"
+                )
```

Comment:
This section does not like me 

```log
boost/1.80.0: Exported revision: df465c37ceb1d7766e55d739265db246
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

bzip2/1.0.8: Main binary package '5eea6cd0d4094bfb10578864f66812e7c6a4d0e6' missing. Using compatible package 'f2848f95be36a5d770fc8e833993b621f703b507'
libbacktrace/cci.20210118: Main binary package '299aaabcb5fb1385c096ccf2c550a653c4438173' missing. Using compatible package 'e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc'
libiconv/1.17: Main binary package '299aaabcb5fb1385c096ccf2c550a653c4438173' missing. Using compatible package 'e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc'
zlib/1.2.13: Main binary package '299aaabcb5fb1385c096ccf2c550a653c4438173' missing. Using compatible package 'e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc'
boost/1.80.0 (test package): Installing package
Requirements
    boost/1.80.0 from local cache - Cache
    bzip2/1.0.8 from 'conancenter' - Cache
    libbacktrace/cci.20210118 from 'conancenter' - Cache
    libiconv/1.17 from 'conancenter' - Cache
    zlib/1.2.13 from 'conancenter' - Cache
Packages
    boost/1.80.0:INVALID - Invalid
    bzip2/1.0.8:f2848f95be36a5d770fc8e833993b621f703b507 - Cache
    libbacktrace/cci.20210118:e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc - Cache
    libiconv/1.17:e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc - Cache
    zlib/1.2.13:e4fbacdee83eb1a2e68227ffe19f6e9a61b36cbc - Cache

Installing (downloading, building) binaries...
ERROR: There are invalid packages (packages that cannot exist for this configuration):
boost/1.80.0: Invalid ID: Boost.{fiber,json,math,nowide,wave} requires a c++11 compiler (please set compiler.cppstd or use a newer compiler)
```

Reply:
The old code had an extra else https://github.com/conan-io/conan-center-index/pull/14415/files#diff-cc14160728d1ad321c1444ffad82615139181e681f4acb7dfd045a005cc8bd66L464, this should nto run with cppstd set

Reply:
Indeed, I've restored the `else`.